### PR TITLE
Controller specs

### PR DESCRIPTION
**Pull request by jhonmunoz**
### Spec generated for Book Controller:
Examples cases:
-`Index `method: Expecting the corresponding Json structure and a 200 HTTP status code.
-`Show` method: Expecting the corresponding Json structure and a 200 HTTP status code.

Both methods expects an authenticated user.
- Authenticated User Context generated

### CODE REVIEW

File: spec/controllers/api/v1/books_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          end
+
+          it 'responses with status 200' do
+            expect(response).to have_http_status(:ok)
+          end
+        end
+      end
+
+      describe 'GET #show' do
+        context 'When fetching a Book' do
```
Comment (petaalcala) [inline spec/controllers/api/v1/books_controller_spec.rb]: Be careful, we only have here the "good" tests.
Do not forget to make the "bad" ones, like testing when asking for a not existing book. They are as important as the anothers.

File: spec/controllers/api/v1/books_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'support/authenticated_user'
```
Comment (petaalcala) [inline spec/controllers/api/v1/books_controller_spec.rb]: I'm not pretty sure, but if you decomment the line 27 of `rails_helper.rb`, you will not need to add this require.https://github.com/wolox-training/jm-rails/blob/839dd8798b66c02a85cae49ced702aedb9e9c619/spec/rails_helper.rb#L27

**Review by petaalcala — CHANGES_REQUESTED** [review]
Just those things.

Great job :)

File: spec/controllers/api/v1/books_controller_spec.rb (hunk 1 above)
Comment (jhonmunoz) [reply spec/controllers/api/v1/books_controller_spec.rb]: Understood, I will add the "bad" scenarios to this test

File: spec/controllers/api/v1/books_controller_spec.rb (hunk 2 above)
Comment (jhonmunoz) [reply spec/controllers/api/v1/books_controller_spec.rb]: You're correct, fixing this...

File: lib/error/error_handler.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      clazz.class_eval do
+        rescue_from ActiveRecord::RecordNotFound, with: :record_not_found
+      end
+    end
+
+    private
+
+    def record_not_found(error)
+      json = Helpers::Render.json(:record_not_found, 404, error.to_s)
+      render json: json, status: 404
```
Comment (petaalcala) [inline lib/error/error_handler.rb]: we tend to use `:not_found` insted of `404`

**Review by hdf1996 — APPROVED** [review]

**Review by petaalcala — APPROVED** [review]

**Pull request merged**